nav2_collision_monitor/src/polygon.cpp

```diff
@@ -229,6 +230,28 @@ int Polygon::getPointsInside(const std::vector<Point> & points) const
   return num;
 }
 
+void Polygon::newDecisionPoint(bool is_in)
+{
+  if (is_in && current_consecutive_in_ < min_consecutive_in_) {
+    current_consecutive_in_++;
+    current_consecutive_out_ = 0;
+    if (current_consecutive_in_ >= min_consecutive_in_) {
+      current_decision_ = true;
```

Comment:
I think the current state needs to be reset on `activate` so that you can activate / deactivate and have it work without stale measurements